IRSR24-22 | @jdwjdwjdw | Add fog and light fog section variants

jdwjdwjdw opened the pull request:
# READY FOR REVIEW

# Summary
- [IRSR24-22](https://stanfordits.atlassian.net/browse/IRSR24-22): Create section variants
  - Fog and Light Fog background-color section variants
![image](https://github.com/SU-SWS/ppo_irsr_subtheme/assets/49299083/e07a501c-1492-4d1d-b091-2f48f16c7429)

# Review By (Date)
- When convenient

# Urgency
- Normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Add a basic page and test out the section background color variants. Confirm they look correct and match [Figma](https://www.figma.com/file/C28Tjq613J2ZK1OmzXdMkB/%5BIRSR%5D-Project-Starter-File?node-id=966%3A13933&mode=dev).
3. Review code 🌁 

# Associated Issues and/or People
- [IRSR24-22](https://stanfordits.atlassian.net/browse/IRSR24-22): Create section variants

[IRSR24-22]: https://stanfordits.atlassian.net/browse/IRSR24-22?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[IRSR24-22]: https://stanfordits.atlassian.net/browse/IRSR24-22?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jdwjdwjdw committed:
Add fog and light fog section variants

jdwjdwjdw committed:
Label fixup

jdwjdwjdw committed:
Update spacing

rebeccahongsf reviewed (changes requested):
So close! Unfortunately, each `.paragraph-item` has a margin-bottom attached to it which causes a white space between two colored section backgrounds.

```
.content .paragraph-item {
  margin-bottom: 7.2rem // equivalent to @include responsive-spacing('margin-bottom', 5)
}
``` 

![image](https://github.com/SU-SWS/ppo_irsr_subtheme/assets/34019925/75ac09ed-65c3-4eed-80a4-9a2456fb49b8)

The immediate workaround that comes to mind would be to remove the margin-bottom and add the spacing to the first nested div, similar to what you did within `_layout.scss` of this PR.

```
.content .paragraph-item {
  margin-bottom: 0;
}

.content .paragraph-item .ptype-stanford-layout {
  @include responsive-spacing('padding-top', 5);
  @include responsive-spacing('padding-bottom', 7);
}
```
However, if you add the above snippet to `layout.scss` then you wouldn't need to apply it again for `.irsr-section-color--foggy` and `.irsr-section-color--foggy-light` so you could remove the following:

```
.irsr-section-color--foggy,
.irsr-section-color--foggy-light {
  @include responsive-spacing('padding-top', 5);
  @include responsive-spacing('padding-bottom', 7);
  ...
}
```

Let me know if you have any follow up questions or concerns 😃

jdwjdwjdw committed:
Update layout styles, fixups

jdwjdwjdw commented:
Thanks @rebeccahongsf! I implemented your suggestion, reorganized/cleaned up the code a bit, and added fixups for pages with side menus and teaser paragraph spacing. Should be good to re-review again.

rebeccahongsf reviewed (approved):
LGTM! Thank you for implementing those changes, @jdwjdwjdw 🎉
